drivers/flash/flash_realtek_rts5912.c

```diff
@@ -0,0 +1,836 @@
+	SPIC_STATUS_t status;
+
+	uint32_t len = 1;
+	uint8_t sr;
+
+	config_command(command, FLASH_CMD_RDSR2, 0, 0, 0);
+	status = spic_read(command, &sr, &len);
+	LOG_DBG("flash: read_sr2 = 0x%02x", sr);
+	if (status != SPIC_STATUS_OK) {
+		return (int)status;
+	}
+
+	*val = sr;
+
+	return 0;
+}
+#endif
+
+static int32_t flash_wait_till_ready(const struct device *dev)
+{
+	int ret;
+	int8_t sr = 0;
+
+	do {
+		ret = (int8_t)flash_read_sr(dev, &sr);
+		if (ret < 0) {
+			return ret;
+		}
+	} while (sr & SR_WIP);
+
+	return 0;
+}
+
+#ifdef CONFIG_FLASH_EX_OP_ENABLED
+static int flash_write_status_reg(const struct device *dev, uint8_t *val, uint8_t cnt)
+{
+	struct flash_rts5912_dev_data *data = dev->data;
+	SPIC_COMMAND_t *command = &data->command_default;
+	SPIC_STATUS_t status;
+
+	uint32_t len = cnt;
+
+	flash_write_enable(dev);
+
+	LOG_DBG("WSR = 0x%02x", *(uint8_t *)val);
+
+	config_command(command, FLASH_CMD_WRSR, 0, 0, 0);
+	status = spic_write(command, val, &len);
+	if (status != SPIC_STATUS_OK) {
+		return (int)status;
+	}
+
+	flash_wait_till_ready(dev);
+	flash_write_disable(dev);
+
+	return 0;
+}
+
+static int flash_write_status_reg2(const struct device *dev, uint8_t *val, uint8_t cnt)
+{
+	struct flash_rts5912_dev_data *data = dev->data;
+	SPIC_COMMAND_t *command = &data->command_default;
+	SPIC_STATUS_t status;
+
+	uint32_t len = cnt;
+
+	flash_write_enable(dev);
+
+	LOG_DBG("WSR2 = 0x%02x", *(uint8_t *)val);
+
+	config_command(command, FLASH_CMD_WRSR2, 0, 0, 0);
+	status = spic_write(command, val, &len);
+	if (status != SPIC_STATUS_OK) {
+		return (int)status;
+	}
+
+	flash_wait_till_ready(dev);
+	flash_write_disable(dev);
+
+	return 0;
+}
+#endif
+
+static int flash_erase_sector(const struct device *dev, uint32_t address)
+{
+	struct flash_rts5912_dev_data *data = dev->data;
+	SPIC_COMMAND_t *command = &data->command_default;
+	SPIC_ADDRESS_SIZE_t addr_size = SPIC_CFG_ADDR_SIZE_24;
+	SPIC_STATUS_t status;
+
+	uint8_t erase_cmd = FLASH_CMD_SE;
+
+	int32_t rc = 0;
+	uint32_t len = 0;
+
+	flash_write_enable(dev);
+
+	config_command(command, erase_cmd, address, addr_size, 0);
+	status = spic_write(command, NULL, &len);
+	if (status != SPIC_STATUS_OK) {
+		rc = -1;
+		goto err_exit;
+	}
+
+	flash_wait_till_ready(dev);
+
+err_exit:
+	flash_write_disable(dev);
+	return rc;
+}
+
+int32_t flash_program_page(const struct device *dev, uint32_t address, const uint8_t *data,
+			   uint32_t size)
+{
+	struct flash_rts5912_dev_data *dev_data = dev->data;
+	SPIC_COMMAND_t *command = &dev_data->command_default;
+	SPIC_ADDRESS_SIZE_t addr_size = SPIC_CFG_ADDR_SIZE_24;
+	SPIC_STATUS_t status;
+
+	uint8_t wr_cmd = FLASH_CMD_PP;
+
+	uint32_t offset = 0, chunk = 0, page_size = FLASH_PAGE_SZ;
+	int32_t rc = 0;
+
+	while (size > 0) {
+		offset = address % page_size;
+		chunk = (offset + size < page_size) ? size : (page_size - offset);
+
+		flash_write_enable(dev);
+
+		config_command(command, wr_cmd, address, addr_size, 0);
```

Comment:
You don't need to call `config_command()` every pass in the loop.  You only need to update the `command->address` on each pass.